pints/_mcmc/_relativistic.py

```diff
@@ -0,0 +1,437 @@
+            self._position = np.array(self._current, copy=True)
+            self._gradient = np.array(self._current_gradient, copy=True)
+            self._momentum = np.array(self._current_momentum, copy=True)
+
+            # Perform a half-step before starting iteration 0 below
+            self._momentum -= self._scaled_epsilon * self._gradient * 0.5
+
+        # Perform a leapfrog step for the position
+        self._position += (
+            self._scaled_epsilon * (self._relativistic_mass()**(-1)) *
+            self._momentum)
+
+        # Ask for the pdf and gradient of the current leapfrog position
+        # Using this, the leapfrog step for the momentum is performed in tell()
+        self._ready_for_tell = True
+        return np.array(self._position, copy=True)
+
+    def current_log_pdf(self):
+        """ See :meth:`SingleChainMCMC.current_log_pdf()`. """
+        return -self._current_energy
+
+    def divergent_iterations(self):
+        """
+        Returns the iteration number of any divergent iterations
+        """
+        return self._divergent
+
+    def epsilon(self):
+        """
+        Returns epsilon used in leapfrog algorithm
+        """
+        return self._epsilon
+
+    def hamiltonian_threshold(self):
+        """
+        Returns threshold difference in Hamiltonian value from one iteration to
+        next which determines whether an iteration is divergent.
+        """
+        return self._hamiltonian_threshold
+
+    def leapfrog_steps(self):
+        """
+        Returns the number of leapfrog steps to carry out for each iteration.
+        """
+        return self._n_frog_iterations
+
+    def leapfrog_step_size(self):
+        """
+        Returns the step size for the leapfrog algorithm.
+        """
+        return self._step_size
+
+    def _log_init(self, logger):
+        """ See :meth:`Loggable._log_init()`. """
+        logger.add_float('Accept.')
+
+    def _log_write(self, logger):
+        """ See :meth:`Loggable._log_write()`. """
+        logger.log(self._mcmc_acceptance)
+
+    def _K(self, momentum):
+        """
+        Kinetic energy or relativistic particle, which is defined in [1]_.
+        """
+        squared = np.sum(np.array(momentum)**2)
+        return self._mc2 * (squared / self._mc2 + 1)**0.5
+
+    def mass(self):
+        """ Returns mass. """
+        return self._mass
+
+    def n_hyper_parameters(self):
+        """ See :meth:`TunableMethod.n_hyper_parameters()`. """
+        return 4
+
+    def name(self):
+        """ See :meth:`pints.MCMCSampler.name()`. """
+        return 'Relativistic MCMC'
+
+    def needs_sensitivities(self):
+        """ See :meth:`pints.MCMCSampler.needs_sensitivities()`. """
+        return True
+
+    def _relativistic_mass(self):
+        """ Calculates relativistic mass. """
+        squared = np.sum(np.array(self._momentum)**2)
+        return self._mass * (squared / self._mc2 + 1)**0.5
```

Comment:
There's a chance the `sqrt()` method will be slightly faster / more accurate, but not a deal breaker!

Reply:
Yep, changed.